Benja stripe doc and clean 1119 1111

Shriukan33 opened the pull request:
Code quality improvement PR

Stripe code has revealed some ill-documented bits, it was time for a bit of cleanup.

This PR improves error handling and makes clearer the intents and what's expected as outputs and inputs.

Closes #1119
Closes #1111

JeffAbrahamson commented on transport_nantes/stripe_app/views.py:
```diff
@@ -380,28 +397,15 @@ def stripe_webhook(request):
             "Details attached to event : \n\n" + "=" * 30 + f"\n{event}"
         )
-        return HttpResponse(status=500)
+        return HttpResponse(status=200)
```
The next line returns a 200 anyway.  Just do it once.

JeffAbrahamson reviewed (changes requested):
One minor change (return within if that should be outside).

The tests change very little despite some rather significant view changes.  Does this signal that the test coverage is inadequate?

Shriukan33 commented:
Although the diff might seem important, it's not that much functionally. 

However, I heard you and expanded the tests to validate that what I changed does what I intended. 

Also fixed the return :)

JeffAbrahamson commented:
This commit comment says the same thing twice:
```
commit 86cfbc4a4060e8a4b1236781732c4b4bdc79e313
Author: Benjamin Mourgues <benjamin@mobilitains.fr>
Date:   Tue Jan 3 10:02:29 2023 +0100

    Replace home-made get_random_string with django's get_random_string
    
    One is maintained by django, the other is not.
    
    Closes #1119
```

I think the point is just to note that they have the same functionality and so this commit is not expected to lead to behavioural change.

JeffAbrahamson commented:
Aside: you should really avoid mixing issues in PRs.  In a professional setting, it would never pass.  Nor in quite a few open source projects.

JeffAbrahamson commented on transport_nantes/stripe_app/static/stripe_app/form_flow.js:
```diff
@@ -323,18 +323,21 @@ fetch("/donation/config/")
+            if (!result.ok) {
+                // Stops the redirection to Stripe Checkout if the server
+                // is not responding OK
+                throw new Error("Erreur lors de l'initialisation de la session de paiement")
```
What catches this error?
What is the user experience if it happens?

Shriukan33 replied:
On the server's end, it's there : 
https://github.com/transport-nantes/tn_web/blob/3dc1d180a7143675fe2089c3d8a96df6347654fb/transport_nantes/stripe_app/views.py#L211-L215

The 500 is not considered OK (2XX are)

Form validation happens before 

https://github.com/transport-nantes/tn_web/blob/3dc1d180a7143675fe2089c3d8a96df6347654fb/transport_nantes/stripe_app/static/stripe_app/form_flow.js#L232-L239

But let's admit validation did pass, but the problem is stripe API key for example, then hitting submit wouldn't try to redirect you to a checkout session, as none has been created.

JeffAbrahamson commented on transport_nantes/stripe_app/static/stripe_app/form_flow.js:
```diff
@@ -323,18 +323,21 @@ fetch("/donation/config/")
-        // Redirect to Stripe Checkout once the checkout session is created
-        return stripe.redirectToCheckout({sessionId: data.sessionId})
+            // Manually trigger before unload event to save success
+            // redirection doesn't seem to trigger it otherwise
```
Avoid using the word "seem" in professional contexts.  It suggests you couldn't be bothered to look further...

JeffAbrahamson reviewed (approved):
LGTM.
